test/app/lib/cinematic/AbstractCommand.spec.js

```diff
@@ -29,23 +41,48 @@ describe('AbstractCommand', () => {
   })
 
   it('run must return a promise', () => {
-    const invalidRunFunctions = [
-      () => undefined,
-      () => true,
-      () => false,
-      () => '',
-      () => 'abc',
-      () => {},
-      () => ({ then: () => { } }), // Fake promise
-      () => Promise.resolve() // non cancellable promise
-    ]
-
-    for (const fn of invalidRunFunctions) {
+    for (const fn of simpleRunFunctions) {
       expect(() => (new PassInRunCommand(fn))[run]()).toThrow()
     }
   })
 })
 
+describe('SyncFunction', () => {
+  it('runs any function passed in', () => {
+    for (const fn of simpleRunFunctions) {
+      expect(() => (new SyncFunction(fn))[run]()).not.toThrow()
+    }
+  })
+
+  it('not running command doesnt call function', () => {
+    const functionSpy = jasmine.createSpy()
+    const command = new SyncFunction(functionSpy)
+    expect(functionSpy).not.toHaveBeenCalled()
+    command[run]()
+    expect(functionSpy).toHaveBeenCalledTimes(1)
+  })
+})
+
+describe('SequentialCommands', () => {
+  it('runs commands sequentially', done => {
+    const functionSpy = jasmine.createSpy()
+    const commands = [
+      () => {
+        expect(functionSpy).not.toHaveBeenCalled()
+      },
+      functionSpy,
+      () => {
+        expect(functionSpy).toHaveBeenCalledTimes(1)
+      },
+      done
+    ]
+
+    const commands2 = commands.map(f => new SyncFunction(f));
+
+    (new SequentialCommands(commands2))[run]()
+  })
+})
+
```

Comment:
Do we also need tests for ConcurrentCommands?

Reply:
I know you reviewed well because you saw this omission.
I have no idea how to test ConcurrentCommands. It's like a concurrent nightmare I have sometimes.

I would love to discuss what you think are good features to test. A simpler example of the same concept is how would you test `Promise.all`.
I will think about this because I think this is a necessary test.

Reply:
Yeah it seems difficult to test. I think the basic thing that we can test for now is to check that it runs the commands and resolves if every command is executed successfully, or gives an error if anyone fails. Does that sound right?

Reply:
That does sound right. I'll add the case that they all run.
I'll also add the  case that they all cancel.
I can write an error test. However the error should just bubble up and throw as we don't catch them.

Thanks for the excellent suggestions.